pgxn/neon/communicator.c

```diff
@@ -526,6 +540,19 @@ communicator_prefetch_pump_state(void)
 
 	END_PREFETCH_RECEIVE_WORK();
 
+	if (RecoveryInProgress())
+	{
+		/*
+		 * Update backend's min in-flight prefetch LSN.
+		 */
+		XLogRecPtr min_backend_prefetch_lsn = GetXLogReplayRecPtr(NULL);
+		for (uint64_t ring_index = MyPState->ring_receive; ring_index < MyPState->ring_unused; ring_index++)
+		{
+			PrefetchRequest* slot = GetPrfSlot(ring_index);
+			min_backend_prefetch_lsn = Min(slot->request_lsns.request_lsn, min_backend_prefetch_lsn);
+		}
+		MIN_BACKEND_PREFETCH_LSN = min_backend_prefetch_lsn;
+	}
```

Comment:
We still need a check for "in-flight" requests, because page_server_request() receives a request with an LSN that may have been generated before pump_state(). 

If we create a `static comm_page_server_request_lsn` that holds the recently called `page_server_request()` request's LSN (and is reset on error/exit), and take that value into account here, then that should fix the issue.